tests/mock_env.py

```diff
@@ -0,0 +1,207 @@
+
+        plugins_conf = [self._make_plugin_conf(plugin_key, args)]
+        setattr(self.workflow, self._plugins_for_phase[phase], plugins_conf)
+
+        return self
+
+    def set_scratch(self, scratch):
+        """
+        Set "scratch" user param to specified value
+        """
+        self.workflow.user_params['scratch'] = scratch
+        return self
+
+    def set_isolated(self, isolated):
+        """
+        Set "isolated" user param to specified value
+        """
+        self.workflow.user_params['isolated'] = isolated
+        return self
+
+    def make_orchestrator(self, orchestrator_args=None):
+        """
+        Make plugin think it is running in orchestrator
+
+        :param orchestrator_args: dict, optional orchestrate_build plugin arguments
+        """
+        if self.workflow.buildstep_plugins_conf:
+            raise ValueError("Buildstep plugin already configured, cannot make orchestrator")
+        self.workflow.buildstep_plugins_conf = [
+            self._make_plugin_conf(PLUGIN_BUILD_ORCHESTRATE_KEY, orchestrator_args)
+        ]
+        return self
+
+    def set_plugin_result(self, phase, plugin_key, result):
+        """
+        Set result of the specified plugin (stored in workflow)
+
+        :param phase: str, plugin phase
+        :param plugin_key: str, plugin key
+        :param result: any, value to set as plugin result
+        """
+        self._validate_phase(phase)
+        results = getattr(self.workflow, self._results_for_phase[phase])
+        results[plugin_key] = result
+        return self
+
+    def set_plugin_args(self, args, phase=None, plugin_key=None):
+        """
+        Set plugin arguments (stored in plugins configuration in workflow).
+
+        By default, sets args for the current plugin (configured using for_plugin()).
+        Phase and plugin key can be specified to set args for a different plugin.
+
+        If overriding phase and plugin key, the specified plugin must already be present
+        in the plugins configuration. Typically, only the current plugin and the
+        orchestrate_build plugin (after make_orchestrator()) will be present.
+
+        :param args: dict, arguments for plugin
+        :param phase: str, optional plugin phase
+        :param plugin_key: str, optional plugin key
+        """
+        phase = phase or self._phase
+        plugin_key = plugin_key or self._plugin_key
+        plugin = self._get_plugin(phase, plugin_key)
+        plugin['args'] = args
+        return self
+
+    def set_reactor_config(self, config):
+        """
+        Set reactor config map (in the ReactorConfigPlugin's workspace).
+
+        :param config: dict or ReactorConfig, if dict, will be converted to ReactorConfig
+        """
+        if not isinstance(config, ReactorConfig):
+            config = ReactorConfig(config)
+        workspace = self._get_reactor_config_workspace()
+        workspace[WORKSPACE_CONF_KEY] = config
+        return self
```

Comment:
Is this correct? `return self`?

Reply:
For method chaining?!

Reply:
@ben-alkov 

> For method chaining?!

Exactly! This is essentially the Builder pattern, you can do things like:
```python
runner = (MockEnv()
          .set_reactor_config({'version': 1, 'operator_manifests': {}})
          .make_orchestrator()
          .create_runner())
```

All methods except `create_runner()` somehow modify state and return `self`, `create_runner()` is the "finisher" that takes the current state of the instance and creates a plugins runner from it.